internal/resource/http/http_test.go

```diff
@@ -281,3 +287,158 @@ func TestResourceWriteHandler(t *testing.T) {
+
+		deletReq.Header.Add("x-consul-token", testACLTokenArtistWritePolicy)
+
+		v2ArtistHandler.ServeHTTP(deleteRsp, deletReq)
+
+		require.Equal(t, http.StatusNoContent, deleteRsp.Result().StatusCode)
+
+		var result map[string]any
+		require.NoError(t, json.NewDecoder(deleteRsp.Body).Decode(&result))
+		require.Empty(t, result)
+
+		_, err := client.Read(testutil.TestContext(t), &pbresource.ReadRequest{
+			Id: &pbresource.ID{
+				Type:    demo.TypeV2Artist,
+				Tenancy: demo.TenancyDefault,
+				Name:    "keith-urban",
+			},
+		})
+		require.ErrorContains(t, err, "resource not found")
+	})
+
+	t.Run("should delete a resource with version", func(t *testing.T) {
+		createResource(t, v2ArtistHandler)
+
+		rsp := httptest.NewRecorder()
+		req := httptest.NewRequest("DELETE", "/demo/v2/artist/keith-urban?partition=default&peer_name=local&namespace=default&version=1", strings.NewReader(""))
+
+		req.Header.Add("x-consul-token", testACLTokenArtistWritePolicy)
+
+		v2ArtistHandler.ServeHTTP(rsp, req)
+
+		require.Equal(t, http.StatusNoContent, rsp.Result().StatusCode)
+
+		_, err := client.Read(testutil.TestContext(t), &pbresource.ReadRequest{
+			Id: &pbresource.ID{
+				Type:    demo.TypeV2Artist,
+				Tenancy: demo.TenancyDefault,
+				Name:    "keith-urban",
+			},
+		})
+		require.ErrorContains(t, err, "resource not found")
+	})
+}
+
+func TestResourceHandler_TypeResolutionError(t *testing.T) {
+	type testCase struct {
+		description string
+		request     *http.Request
+		response    *httptest.ResponseRecorder
+	}
+	client := svctest.RunResourceService(t, demo.RegisterTypes)
+	resourceHandler := resourceHandler{
+		resource.Registration{
+			Type: &pbresource.Type{
+				Group:        "test",
+				GroupVersion: "v0",
+				Kind:         "test",
+			},
+			Proto: &pbdemov2.Artist{},
+		},
+		client,
+		func(req *http.Request, token *string) { return },
+		hclog.NewNullLogger(),
+	}
+
+	testCases := []testCase{
+		{
+			description: "write resource",
+			request: httptest.NewRequest("PUT", "/demo/v2/artist/keith-urban?partition=default&peer_name=local&namespace=default", strings.NewReader(`
+				{
+					"metadata": {
+						"foo": "bar"
+					},
+					"data": {
+						"name": "Keith Urban",
+						"genre": "GENRE_COUNTRY"
+					}
+				}
+			`)),
+			response: httptest.NewRecorder(),
+		},
+		{
+			description: "delete resource",
+			request:     httptest.NewRequest("DELETE", "/demo/v2/artist/keith-urban?partition=default&peer_name=local&namespace=default", strings.NewReader("")),
+			response:    httptest.NewRecorder(),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			resourceHandler.ServeHTTP(tc.response, tc.request)
+
+			require.Equal(t, http.StatusBadRequest, tc.response.Result().StatusCode)
+			require.Contains(t, tc.response.Body.String(), "resource type test.v0.test not registered")
+		})
+	}
+}
```

Comment:
I added this test case for the code path [here](https://github.com/hashicorp/consul/blob/main/agent/grpc-external/services/resource/delete.go#L34)
I probably don't need this for the HTTP side since the endpoints are registered dynamically from the registry so registry mismatch is quite unlikely.  WDYT?

Reply:
Yeah, the exposed HTTP endpoints are built on the registered types so this seems unnecessary.